Scripts/Runtime/Jobs/Collections/DynamicBufferSharedWriteHandle.cs

```diff
@@ -0,0 +1,294 @@
+    /// However, there are ways that you won't get the results you expect. They have been listed to aid in debugging
+    /// if needed.
+    ///
+    /// ALL OF THESE SHOULD BE VERY RARE
+    ///
+    /// - If you are adding/removing/reordering Systems and SystemGroups and messing with the PlayerLoop on a frequent
+    /// basis. Ex: Per frame. The <see cref="WorldCache"/> may not update properly and you will have to update manually
+    /// via <see cref="WorldCache.ManualRebuild"/>
+    ///
+    /// - If you are using <see cref="EntityQuery"/>s that your Systems aren't aware of.
+    /// Ex. <see cref="EntityManager.CreateEntityQuery"/> then we may not detect jobs that touch your
+    /// <see cref="IBufferElementData"/>. Always use the <see cref="SystemBase.GetEntityQuery"/> function.
+    ///
+    /// - If you are scheduling jobs outside of a System that operate on <see cref="IBufferElementData"/> we will not
+    /// be aware of it.
+    ///
+    /// - If you are scheduling more than one job in a System that operates on the <see cref="IBufferElementData"/> and
+    /// one uses the Shared Write handle and the other(s) try and do an exclusive write or read, we treat that
+    /// System as only using the Shared Write handle.
+    ///
+    /// - If you are dynamically turning on and off queries (create and/or destroy or the system runs for a while until
+    /// it passes an if check and only THEN you call the <see cref="SystemBase.GetEntityQuery"/> function that operates
+    /// on your <see cref="IBufferElementData"/>, then we will likely not know about it until the
+    /// <see cref="WorldCache"/> is rebuilt. You should do this manually. Better yet, create all possible queries in
+    /// your <see cref="SystemBase.OnCreate"/> so we're aware even if you don't use them until later.
+    ///
+    /// - You may get a false positive if you have a System that has two or more queries in it. One that operates on 
+    /// your <see cref="IBufferElementData"/> (QueryA) and one that doesn't (QueryB). If the logic has the QueryB
+    /// execute but QueryA doesn't, we still see the system as having executed and we'll count it as a point to move
+    /// the handle up when in reality it could have been ignored. 
+    /// 
+    /// </summary>
+    /// <remarks>
+    /// This is similar to the <see cref="CollectionAccessController{TContext}"/> but for specific use with a
+    /// <see cref="DynamicBuffer{T}"/> that other systems are also using for reading and/or writing but might
+    /// not be aware of the access pattern for shared writing.
+    /// <seealso cref="DynamicBufferSharedWriteUtil"/>
+    /// </remarks>
+    /// <typeparam name="T">The <see cref="IBufferElementData"/> type this instance is associated with.</typeparam>
+    public class DynamicBufferSharedWriteHandle<T> : AbstractAnvilBase,
+                                                     DynamicBufferSharedWriteUtil.IDynamicBufferSharedWriteHandle
```

Comment:
Consider refactoring this class a bit to have the instance that the System holds be a lens into the shared access controller from the perspective of the system.
Similar to how `Log.Logger` or `UpdateHandle` work.

This could all be done through a static method on the access controller class like DynamicBufferSharedWriteHandle (pka:`DynamicBufferSharedWriteAccessController.`)
 - Can possibly make all DynamicBufferSharedWriteAccessController methods internal or private then?

**Example Usage:**
``` csharp
/// OnCreate - registers and provides a system specific context
DynamicBufferSharedWriteContext m_PendingHealthWriteContext = GetSharedWriteContext<PendingHealthDB>();

// OnUpdate
JobHandle sharedWrite = m_PendingHealthWriteContext.GetSharedWriteHandle(Dependency);

//OnDestroy - unregisters
m_PendingHealthWriteContext.Dispose();
```

Reply:
Hmmmn ok, I see where you are going with this. I like it.

Reply:
Done. This was a good idea.